<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1055</width>
    <height>650</height>
   </rect>
  </property>
  <property name="documentMode">
   <bool>true</bool>
  </property>
  <property name="dockNestingEnabled">
   <bool>true</bool>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout_4">
    <property name="leftMargin">
     <number>0</number>
    </property>
    <property name="topMargin">
     <number>0</number>
    </property>
    <property name="rightMargin">
     <number>0</number>
    </property>
    <property name="bottomMargin">
     <number>0</number>
    </property>
    <item row="0" column="0">
     <widget class="Screen" name="screen">
      <property name="sizePolicy">
       <sizepolicy hsizetype="MinimumExpanding" vsizetype="MinimumExpanding">
        <horstretch>1</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1055</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>Game</string>
    </property>
    <addaction name="actionOpenAny"/>
    <addaction name="separator"/>
    <addaction name="actionOpen"/>
    <addaction name="actionIntegrate"/>
    <addaction name="actionLoadExisting"/>
    <addaction name="separator"/>
    <addaction name="actionLoadState"/>
    <addaction name="actionSaveState"/>
    <addaction name="actionScreenshot"/>
    <addaction name="separator"/>
    <addaction name="actionReset"/>
    <addaction name="actionHardReset"/>
    <addaction name="actionPause"/>
    <addaction name="actionStep"/>
    <addaction name="separator"/>
    <addaction name="actionPlayMovie"/>
    <addaction name="actionRecordMovie"/>
    <addaction name="actionStopMovie"/>
    <addaction name="actionEpisodeRecording"/>
   </widget>
   <widget class="QMenu" name="menuData">
    <property name="title">
     <string>Data</string>
    </property>
    <addaction name="actionLoadVars"/>
    <addaction name="actionSaveVars"/>
    <addaction name="separator"/>
    <addaction name="actionLoadSearch"/>
    <addaction name="actionSaveSearch"/>
    <addaction name="separator"/>
    <addaction name="actionLoadScenario"/>
    <addaction name="actionEditScenario"/>
    <addaction name="separator"/>
    <addaction name="actionAutosaveVariables"/>
    <addaction name="actionAutosaveScenario"/>
    <addaction name="actionAutosaveSearches"/>
    <addaction name="separator"/>
    <addaction name="actionSetDataDirectory"/>
   </widget>
   <widget class="QMenu" name="menuWindow">
    <property name="title">
     <string>Window</string>
    </property>
    <addaction name="actionShowVariables"/>
    <addaction name="actionShowScenario"/>
    <addaction name="actionShowSearch"/>
    <addaction name="actionShowCheats"/>
    <addaction name="separator"/>
    <addaction name="actionControls"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuData"/>
   <addaction name="menuWindow"/>
  </widget>
  <widget class="QDockWidget" name="variableView">
   <property name="windowTitle">
    <string>Variables</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>2</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents">
    <layout class="QGridLayout" name="gridLayout_2">
     <item row="2" column="0">
      <widget class="QSpinBox" name="variableAddress">
       <property name="accelerated">
        <bool>true</bool>
       </property>
       <property name="minimum">
        <number>0</number>
       </property>
       <property name="maximum">
        <number>2147483647</number>
       </property>
       <property name="displayIntegerBase">
        <number>16</number>
       </property>
      </widget>
     </item>
     <item row="1" column="3">
      <widget class="QToolButton" name="variableRemove">
       <property name="text">
        <string>-</string>
       </property>
      </widget>
     </item>
     <item row="0" column="0" colspan="4">
      <widget class="QTableView" name="variables">
       <property name="alternatingRowColors">
        <bool>true</bool>
       </property>
       <property name="selectionBehavior">
        <enum>QAbstractItemView::SelectRows</enum>
       </property>
       <property name="showGrid">
        <bool>false</bool>
       </property>
       <attribute name="horizontalHeaderDefaultSectionSize">
        <number>60</number>
       </attribute>
       <attribute name="horizontalHeaderHighlightSections">
        <bool>false</bool>
       </attribute>
       <attribute name="horizontalHeaderStretchLastSection">
        <bool>true</bool>
       </attribute>
       <attribute name="verticalHeaderVisible">
        <bool>false</bool>
       </attribute>
      </widget>
     </item>
     <item row="1" column="2">
      <widget class="QToolButton" name="variableAdd">
       <property name="text">
        <string>+</string>
       </property>
      </widget>
     </item>
     <item row="2" column="1" colspan="3">
      <widget class="QLineEdit" name="variableType">
       <property name="maxLength">
        <number>4</number>
       </property>
       <property name="placeholderText">
        <string>Type (e.g. =u4)</string>
       </property>
      </widget>
     </item>
     <item row="1" column="0" colspan="2">
      <widget class="QLineEdit" name="variableName">
       <property name="placeholderText">
        <string>New variable name</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="scenarioView">
   <property name="sizePolicy">
    <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="windowTitle">
    <string>Scenario information</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>2</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_2">
    <layout class="QGridLayout" name="gridLayout_3">
     <item row="0" column="2">
      <widget class="QLabel" name="label_2">
       <property name="text">
        <string>Cumulative:</string>
       </property>
      </widget>
     </item>
     <item row="0" column="0">
      <widget class="QLabel" name="label">
       <property name="text">
        <string>Reward:</string>
       </property>
      </widget>
     </item>
     <item row="0" column="3">
      <widget class="QLabel" name="cumReward">
       <property name="text">
        <string>0</string>
       </property>
      </widget>
     </item>
     <item row="4" column="2" colspan="2">
      <widget class="QPushButton" name="resetScenario">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="text">
        <string>Reset scenario</string>
       </property>
      </widget>
     </item>
     <item row="2" column="3">
      <widget class="QLabel" name="didEnd">
       <property name="text">
        <string>No</string>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QLabel" name="reward">
       <property name="text">
        <string>0</string>
       </property>
      </widget>
     </item>
     <item row="4" column="0">
      <widget class="QLabel" name="label_5">
       <property name="text">
        <string>Movie:</string>
       </property>
      </widget>
     </item>
     <item row="4" column="1">
      <widget class="QLabel" name="movie">
       <property name="text">
        <string>None</string>
       </property>
      </widget>
     </item>
     <item row="2" column="2">
      <widget class="QLabel" name="label_4">
       <property name="text">
        <string>Did end:</string>
       </property>
      </widget>
     </item>
     <item row="2" column="0">
      <widget class="QLabel" name="label_3">
       <property name="text">
        <string>Done:</string>
       </property>
      </widget>
     </item>
     <item row="2" column="1">
      <widget class="QLabel" name="done">
       <property name="text">
        <string>No</string>
       </property>
      </widget>
     </item>
     <item row="3" column="0">
      <widget class="QLabel" name="label_6">
       <property name="text">
        <string>Frame:</string>
       </property>
      </widget>
     </item>
     <item row="3" column="2">
      <widget class="QLabel" name="label_7">
       <property name="text">
        <string>Timestep:</string>
       </property>
      </widget>
     </item>
     <item row="3" column="1">
      <widget class="QLabel" name="frame">
       <property name="text">
        <string>0</string>
       </property>
      </widget>
     </item>
     <item row="3" column="3">
      <widget class="QLabel" name="ts">
       <property name="text">
        <string>0</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="searchView">
   <property name="windowTitle">
    <string>Search</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>2</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_3">
    <layout class="QGridLayout" name="gridLayout" rowstretch="2,3,0,0">
     <item row="2" column="2">
      <widget class="QToolButton" name="searchSubmit">
       <property name="text">
        <string>Search</string>
       </property>
      </widget>
     </item>
     <item row="0" column="0" colspan="4">
      <widget class="QTableView" name="search">
       <property name="alternatingRowColors">
        <bool>true</bool>
       </property>
       <property name="selectionBehavior">
        <enum>QAbstractItemView::SelectRows</enum>
       </property>
       <property name="showGrid">
        <bool>false</bool>
       </property>
       <attribute name="horizontalHeaderDefaultSectionSize">
        <number>60</number>
       </attribute>
       <attribute name="horizontalHeaderHighlightSections">
        <bool>false</bool>
       </attribute>
       <attribute name="horizontalHeaderStretchLastSection">
        <bool>true</bool>
       </attribute>
       <attribute name="verticalHeaderVisible">
        <bool>false</bool>
       </attribute>
      </widget>
     </item>
     <item row="2" column="0" colspan="2">
      <widget class="QLineEdit" name="searchName">
       <property name="placeholderText">
        <string>Search name</string>
       </property>
      </widget>
     </item>
     <item row="3" column="0">
      <widget class="QComboBox" name="searchType">
       <item>
        <property name="text">
         <string>Is</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Increased by</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Decreased by</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Increased</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Decreased</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Changed</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Unchanged</string>
        </property>
       </item>
      </widget>
     </item>
     <item row="2" column="3">
      <widget class="QToolButton" name="searchRemove">
       <property name="text">
        <string>-</string>
       </property>
      </widget>
     </item>
     <item row="1" column="0" colspan="4">
      <widget class="QTableView" name="searchResults">
       <property name="alternatingRowColors">
        <bool>true</bool>
       </property>
       <property name="selectionBehavior">
        <enum>QAbstractItemView::SelectRows</enum>
       </property>
       <property name="showGrid">
        <bool>false</bool>
       </property>
       <attribute name="horizontalHeaderDefaultSectionSize">
        <number>60</number>
       </attribute>
       <attribute name="horizontalHeaderHighlightSections">
        <bool>false</bool>
       </attribute>
       <attribute name="horizontalHeaderStretchLastSection">
        <bool>true</bool>
       </attribute>
       <attribute name="verticalHeaderVisible">
        <bool>false</bool>
       </attribute>
      </widget>
     </item>
     <item row="3" column="1">
      <widget class="QSpinBox" name="searchValue">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="accelerated">
        <bool>true</bool>
       </property>
       <property name="minimum">
        <number>-2147483647</number>
       </property>
       <property name="maximum">
        <number>2147483647</number>
       </property>
      </widget>
     </item>
     <item row="3" column="2" colspan="2">
      <widget class="QToolButton" name="searchMigrate">
       <property name="text">
        <string>Make variable</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="cheatView">
   <property name="visible">
    <bool>false</bool>
   </property>
   <property name="floating">
    <bool>false</bool>
   </property>
   <property name="windowTitle">
    <string>Cheats</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>2</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_4">
    <layout class="QGridLayout" name="gridLayout_5" rowstretch="2,0,1">
     <item row="0" column="0" colspan="3">
      <widget class="QTableView" name="cheats">
       <property name="alternatingRowColors">
        <bool>true</bool>
       </property>
       <property name="selectionBehavior">
        <enum>QAbstractItemView::SelectRows</enum>
       </property>
       <property name="showGrid">
        <bool>false</bool>
       </property>
       <attribute name="horizontalHeaderDefaultSectionSize">
        <number>60</number>
       </attribute>
       <attribute name="horizontalHeaderHighlightSections">
        <bool>false</bool>
       </attribute>
       <attribute name="horizontalHeaderStretchLastSection">
        <bool>true</bool>
       </attribute>
       <attribute name="verticalHeaderVisible">
        <bool>false</bool>
       </attribute>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QLineEdit" name="cheatName">
       <property name="placeholderText">
        <string>New cheat name</string>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <widget class="QToolButton" name="cheatAdd">
       <property name="text">
        <string>+</string>
       </property>
      </widget>
     </item>
     <item row="1" column="2">
      <widget class="QToolButton" name="cheatRemove">
       <property name="text">
        <string>-</string>
       </property>
      </widget>
     </item>
     <item row="2" column="0" colspan="3">
      <widget class="QPlainTextEdit" name="cheatValue">
       <property name="placeholderText">
        <string/>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <action name="actionOpen">
   <property name="text">
    <string>&amp;Load game...</string>
   </property>
   <property name="shortcut">
    <string>Meta+Ctrl+O</string>
   </property>
  </action>
  <action name="actionPause">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>&amp;Pause</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+P</string>
   </property>
  </action>
  <action name="actionStep">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Step</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
  <action name="actionLoadState">
   <property name="text">
    <string>Load state...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+L</string>
   </property>
  </action>
  <action name="actionSaveState">
   <property name="text">
    <string>Save state...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionReset">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Reset</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+R</string>
   </property>
  </action>
  <action name="actionLoadVars">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Load variables...</string>
   </property>
  </action>
  <action name="actionSaveVars">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Save variables...</string>
   </property>
  </action>
  <action name="actionLoadSearch">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Load search...</string>
   </property>
  </action>
  <action name="actionSaveSearch">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Save search...</string>
   </property>
  </action>
  <action name="actionEditScenario">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Edit scenario...</string>
   </property>
  </action>
  <action name="actionScreenshot">
   <property name="text">
    <string>Screenshot...</string>
   </property>
  </action>
  <action name="actionAutosaveVariables">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Autosave variables</string>
   </property>
  </action>
  <action name="actionAutosaveSearches">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Autosave searches</string>
   </property>
  </action>
  <action name="actionAutosaveScenario">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Autosave scenario</string>
   </property>
  </action>
  <action name="actionIntegrate">
   <property name="text">
    <string>Integrate...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+O</string>
   </property>
  </action>
  <action name="actionLoadExisting">
   <property name="text">
    <string>Load existing...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Alt+O</string>
   </property>
  </action>
  <action name="actionShowCheats">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Show cheats...</string>
   </property>
  </action>
  <action name="actionShowVariables">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Show variables...</string>
   </property>
  </action>
  <action name="actionShowScenario">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Show scenario info...</string>
   </property>
  </action>
  <action name="actionShowSearch">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Show search...</string>
   </property>
  </action>
  <action name="actionPlayMovie">
   <property name="text">
    <string>Play movie...</string>
   </property>
  </action>
  <action name="actionRecordMovie">
   <property name="text">
    <string>Record movie...</string>
   </property>
  </action>
  <action name="actionStopMovie">
   <property name="text">
    <string>Stop movie</string>
   </property>
  </action>
  <action name="actionEpisodeRecording">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Episode record mode</string>
   </property>
  </action>
  <action name="actionControls">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Controls...</string>
   </property>
  </action>
  <action name="actionHardReset">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Hard reset</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+R</string>
   </property>
  </action>
  <action name="actionOpenAny">
   <property name="text">
    <string>Open...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionSetDataDirectory">
   <property name="text">
    <string>Set data directory...</string>
   </property>
  </action>
  <action name="actionLoadScenario">
   <property name="text">
    <string>Load scenario...</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>Screen</class>
   <extends>QOpenGLWidget</extends>
   <header>Screen.h</header>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>variables</tabstop>
  <tabstop>variableName</tabstop>
  <tabstop>variableAddress</tabstop>
  <tabstop>variableType</tabstop>
  <tabstop>variableAdd</tabstop>
  <tabstop>variableRemove</tabstop>
  <tabstop>resetScenario</tabstop>
  <tabstop>search</tabstop>
  <tabstop>searchResults</tabstop>
  <tabstop>searchName</tabstop>
  <tabstop>searchType</tabstop>
  <tabstop>searchValue</tabstop>
  <tabstop>searchSubmit</tabstop>
  <tabstop>searchRemove</tabstop>
  <tabstop>searchMigrate</tabstop>
  <tabstop>cheats</tabstop>
  <tabstop>cheatName</tabstop>
  <tabstop>cheatValue</tabstop>
  <tabstop>cheatAdd</tabstop>
  <tabstop>cheatRemove</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>cheats</sender>
   <signal>doubleClicked(QModelIndex)</signal>
   <receiver>screen</receiver>
   <slot>gamesWereBetter()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1147</x>
     <y>708</y>
    </hint>
    <hint type="destinationlabel">
     <x>479</x>
     <y>451</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionShowVariables</sender>
   <signal>toggled(bool)</signal>
   <receiver>variableView</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>1147</x>
     <y>117</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>variableView</sender>
   <signal>visibilityChanged(bool)</signal>
   <receiver>actionShowVariables</receiver>
   <slot>setChecked(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1147</x>
     <y>117</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>scenarioView</sender>
   <signal>visibilityChanged(bool)</signal>
   <receiver>actionShowScenario</receiver>
   <slot>setChecked(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1147</x>
     <y>282</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>searchView</sender>
   <signal>visibilityChanged(bool)</signal>
   <receiver>actionShowSearch</receiver>
   <slot>setChecked(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1147</x>
     <y>492</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cheatView</sender>
   <signal>visibilityChanged(bool)</signal>
   <receiver>actionShowCheats</receiver>
   <slot>setChecked(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1147</x>
     <y>757</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionShowScenario</sender>
   <signal>toggled(bool)</signal>
   <receiver>scenarioView</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>1147</x>
     <y>282</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionShowSearch</sender>
   <signal>toggled(bool)</signal>
   <receiver>searchView</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>1147</x>
     <y>492</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionShowCheats</sender>
   <signal>toggled(bool)</signal>
   <receiver>cheatView</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>1147</x>
     <y>757</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
